mmdet3d/models/roi_heads/bbox_heads/multi_group_head.py

```diff
@@ -0,0 +1,868 @@
+        heatmaps, anno_boxes, inds, masks = [], [], [], []
+
+        for idx, task in enumerate(self.tasks):
+            heatmap = gt_bboxes_3d.new_zeros(
+                (len(self.class_names[idx]), feature_map_size[1],
+                 feature_map_size[0]))
+
+            anno_box = gt_bboxes_3d.new_zeros((max_objs, 10),
+                                              dtype=torch.float32)
+
+            ind = gt_labels_3d.new_zeros((max_objs), dtype=torch.int64)
+            mask = gt_bboxes_3d.new_zeros((max_objs), dtype=torch.uint8)
+
+            num_objs = min(task_boxes[idx].shape[0], max_objs)
+
+            for k in range(num_objs):
+                cls_id = task_classes[idx][k] - 1
+
+                width, length, _ = task_boxes[idx][k][3], task_boxes[idx][k][
+                    4], task_boxes[idx][k][5]
+                width, length = width / voxel_size[0] / self.train_cfg[
+                    'out_size_factor'], length / voxel_size[
+                        1] / self.train_cfg['out_size_factor']
+                if width > 0 and length > 0:
+                    radius = gaussian_radius(
+                        (length, width),
+                        min_overlap=self.train_cfg['gaussian_overlap'])
+                    radius = max(self.train_cfg['min_radius'], int(radius))
+
+                    # be really careful for the coordinate system of
+                    # your box annotation.
+                    x, y, z = task_boxes[idx][k][0], task_boxes[idx][k][
+                        1], task_boxes[idx][k][2]
+
+                    coor_x, coor_y = (
+                        x - pc_range[0]
+                    ) / voxel_size[0] / self.train_cfg['out_size_factor'], (
+                        y - pc_range[1]
+                    ) / voxel_size[1] / self.train_cfg['out_size_factor']
+
+                    ct = torch.tensor([coor_x, coor_y],
+                                      dtype=torch.float32,
+                                      device=device)
+                    ct_int = ct.to(torch.int32)
+
+                    # throw out not in range objects to avoid out of array
+                    # area when creating the heatmap
+                    if not (0 <= ct_int[0] < feature_map_size[0]
+                            and 0 <= ct_int[1] < feature_map_size[1]):
+                        continue
+
+                    draw_gaussian(heatmap[cls_id], ct_int, radius)
+
+                    new_idx = k
+                    x, y = ct_int[0], ct_int[1]
+
+                    assert (y * feature_map_size[0] + x <
+                            feature_map_size[0] * feature_map_size[1])
+
+                    ind[new_idx] = y * feature_map_size[0] + x
+                    mask[new_idx] = 1
+                    # TODO: support other outdoor dataset
+                    vx, vy = task_boxes[idx][k][6:8]
+                    rot = task_boxes[idx][k][8]
+                    if not self.train_cfg['no_log']:
```

Comment:
Make `self.train_cfg['no_log']` -> `self.norm_bbox`, since `no_log` is common option used in both training and testing.